"numpy.ndarray size changed" error when installing `pydivsufsort` together with `tensorflow==2.6.0`

There seems to be some incompatibility between the C-interface of `pydivsufsort` and that of numpy. 

The observations are:
```
/usr/local/lib/python3.8/site-packages/mypack/test.py:5: in <module>
    from pydivsufsort import divsufsort, kasai
/usr/local/lib/python3.8/site-packages/pydivsufsort/__init__.py:2: in <module>
    from .stringalg import (
pydivsufsort/stringalg.pyx:1: in init pydivsufsort.stringalg
    ???
E   ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 88 from C header, got 80 from PyObject
```

After some extensive searching on the internet, I have found out it is likely related to the `numpy`'s change of the C-interface:
https://stackoverflow.com/questions/66060487/valueerror-numpy-ndarray-size-changed-may-indicate-binary-incompatibility-exp
And in particular this answer, which also avoid updating `numpy` (it is imposed by my Tensorflow):
https://stackoverflow.com/a/69528768/2470337

So I was wondering if you have any idea of what this error may mean and how it could be corrected?
<details>
  <summary>Ubuntu`pip freeze`</summary>

```
absl-py==0.15.0
astunparse==1.6.3
cachetools==4.2.4
certifi==2021.10.8
charset-normalizer==2.0.7
clang==5.0
eccodes==1.3.3
flatbuffers==1.12
gast==0.4.0
google-auth==2.3.2
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.41.1
h5py==3.1.0
idna==3.3
keras==2.6.0
Keras-Preprocessing==1.1.2
Markdown==3.3.4
numpy==1.19.5
oauthlib==3.1.1
opt-einsum==3.3.0
protobuf==3.19.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pydivsufsort==0.0.5
requests==2.26.0
requests-oauthlib==1.3.0
rsa==4.7.2
six==1.15.0
tensorboard==2.7.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.6.0
tensorflow-estimator==2.6.0
termcolor==1.1.0
typing-extensions==3.7.4.3
urllib3==1.26.7
Werkzeug==2.0.2
wrapt==1.12.1
```
</details>

I only have this problem on Ubuntu systems. On Windows everything works just fine. My Windows `pip freeze`:
<details>
  <summary>Windows `pip freeze`</summary>

```
absl-py==0.15.0
appdirs==1.4.4
argon2-cffi==21.1.0
astroid==2.8.4
astunparse==1.6.3
atomicwrites==1.4.0
attrs==21.2.0
autopep8==1.6.0
backcall==0.2.0
backports.entry-points-selectable==1.1.0
bleach==4.1.0
cachetools==4.2.4
certifi==2021.10.8
cffi==1.15.0
cfgrib==0.9.9.0
cfgv==3.3.1
cftime==1.5.1
charset-normalizer==2.0.7
clang==5.0
click==8.0.3
colorama==0.4.4
cycler==0.10.0
debugpy==1.5.1
decorator==5.1.0
defusedxml==0.7.1
dill==0.3.4
distlib==0.3.3
eccodes==1.3.3
emcee==3.1.1
entrypoints==0.3
et-xmlfile==1.1.0
execnet==1.9.0
filelock==3.3.1
findlibs==0.0.2
flatbuffers==1.12
func-timeout==4.3.5
gast==0.4.0
google-auth==2.3.1
google-auth-oauthlib==0.4.6
google-pasta==0.2.0
grpcio==1.41.1
h5py==3.1.0
identify==2.3.1
idna==3.3
importlib-metadata==4.8.1
iniconfig==1.1.1
ipykernel==6.4.2
ipython==7.28.0
ipython-genutils==0.2.0
ipywidgets==7.6.5
iso8601==0.1.16
isort==5.9.3
jedi==0.18.0
Jinja2==3.0.2
joblib==1.1.0
jsonschema==4.1.2
julian==0.14
jupyter-client==7.0.6
jupyter-core==4.8.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.2
kaleido==0.2.1
keras==2.6.0
Keras-Preprocessing==1.1.2
kiwisolver==1.3.2
lazy-object-proxy==1.6.0
lz4==3.1.3
Markdown==3.3.4
MarkupSafe==2.0.1
marshmallow==3.14.0
matplotlib==3.4.3
matplotlib-inline==0.1.3
mccabe==0.6.1
mistune==0.8.4
more-itertools==8.10.0
multiprocess==0.70.12.2
nbclient==0.5.4
nbconvert==6.2.0
nbformat==5.1.3
nest-asyncio==1.5.1
netCDF4==1.5.7
nodeenv==1.6.0
nose==1.3.7
notebook==6.4.5
numpy==1.19.5
oauthlib==3.1.1
openpyxl==3.0.9
opt-einsum==3.3.0
packaging==21.0
pandas==1.3.2
pandocfilters==1.5.0
parso==0.8.2
pathlib==1.0.1
pathos==0.2.8
patsy==0.5.2
pickleshare==0.7.5
Pillow==8.4.0
pipdeptree==2.2.0
platformdirs==2.4.0
plotly==5.3.1
pluggy==1.0.0
ply==3.11
pox==0.3.0
ppft==1.6.6.4
pre-commit==2.15.0
prometheus-client==0.11.0
prompt-toolkit==3.0.21
protobuf==3.19.0
py==1.10.0
pyarrow==4.0.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycodestyle==2.8.0
pycparser==2.20
pydivsufsort==0.0.5
Pygments==2.10.0
pylint==2.11.1
Pympler==0.9
Pyomo==6.1.2
pyparsing==3.0.1
pyrsistent==0.18.0
pytest==6.2.5
pytest-forked==1.3.0
pytest-xdist==2.4.0
python-dateutil==2.8.2
pytz==2021.3
PyUtilib==6.0.0
pywin32==302
pywinpty==1.1.4
PyYAML==6.0
pyzmq==22.3.0
requests==2.26.0
requests-oauthlib==1.3.0
retrying==1.3.3
rsa==4.7.2
scikit-learn==1.0.1
scipy==1.7.1
seaborn==0.10.1
Send2Trash==1.8.0
singledispatchmethod==1.0
six==1.15.0
statsmodels==0.13.0
tenacity==8.0.1
tensorboard==2.7.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
tensorflow==2.6.0
tensorflow-estimator==2.6.0
termcolor==1.1.0
terminado==0.12.1
testpath==0.5.0
threadpoolctl==3.0.0
toml==0.10.2
tornado==6.1
tqdm==4.62.3
traitlets==5.1.1
typing-extensions==3.7.4.3
urllib3==1.26.7
virtualenv==20.9.0
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==2.0.2
widgetsnbextension==3.5.1
wrapt==1.12.1
xarray==0.19.0
xlrd==2.0.1
xmltodict==0.12.0
zipp==3.6.0
```
</details>

--- a/.travis.yml
+++ b/.travis.yml
@@ -2,6 +2,9 @@ notifications:
   email: false
 language: shell
 
+env:
+  - NUMPY_BUILD_VERSION=1.10.4
+
 jobs:
   include:
     # linux build
@@ -9,12 +12,12 @@ jobs:
       language: python
       env:
         - PYTHON=python3
-        - CIBW_BEFORE_BUILD="pip install Cython numpy"
+        - CIBW_BEFORE_BUILD="pip install Cython numpy==$NUMPY_BUILD_VERSION"
     # macos build
     - os: osx
       env:
         - PYTHON=python3
-        - CIBW_BEFORE_BUILD="pip install Cython numpy"
+        - CIBW_BEFORE_BUILD="pip install Cython numpy==$NUMPY_BUILD_VERSION"
     # win_amd64 build
     - os: windows
       before_install:
@@ -24,7 +27,7 @@ jobs:
         - CIBW_BUILD="*-win_amd64"
         - PLATFORM_OPTION="-A x64"
         - PYTHON=/c/Python38/python.exe
-        - CIBW_BEFORE_BUILD="bash build.sh"
+        - CIBW_BEFORE_BUILD="pip install numpy==$NUMPY_BUILD_VERSION; bash build.sh"
     # win32 build
     - os: windows
       before_install:
@@ -33,14 +36,14 @@ jobs:
       env:
         - CIBW_BUILD="*-win32"
         - PYTHON=/c/Python38/python.exe
-        - CIBW_BEFORE_BUILD="bash build.sh"
+        - CIBW_BEFORE_BUILD="pip install numpy==$NUMPY_BUILD_VERSION; bash build.sh"
     - language: python
       arch: arm64
       install:
       script:
         # without --no-cache:
         # [Errno 13] Permission denied: '/home/travis/.cache/pip/wheels/'
-        - python3 -m pip install -U --no-cache setuptools pip wheel Cython
+        - python3 -m pip install -U --no-cache setuptools pip wheel Cython "numpy==$NUMPY_BUILD_VERSION"
         - python3 -m pip install flake8 codecov
         - flake8
         - python3 setup.py sdist

